unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,159 @@
+import json
+from dataclasses import dataclass, field
+from datetime import timedelta
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: Optional[str] = None
+    username: Optional[str] = None
+    password: Optional[str] = None
+
+
+@dataclass
+class CouchbaseConnectionConfig(ConnectionConfig):
+    access_config: CouchbaseAccessConfig = enhanced_field(
+        sensitive=True, default_factory=CouchbaseAccessConfig
+    )
+    bucket: Optional[str] = None
+    scope: Optional[str] = None
+    collection: Optional[str] = None
+    batch_size: int = 50
+    connector_type: str = CONNECTOR_TYPE
+
+    def __post_init__(self):
+        # verify if required fields are present
+        required_fields = ["connection_string", "username", "password"]
+        for cb_field in required_fields:
+            if (not getattr(self.access_config, cb_field)) or getattr(
+                self.access_config, cb_field
+            ) == "":
+                raise ValueError(f"Missing field in couchbase access config: {cb_field}")
```

Comment:
Move this up as a check on the access configs
```suggestion
```

Reply:
Updated